cfme/tests/networks/nuage/test_nuage_inventory.py

```diff
@@ -51,3 +51,39 @@ def subnet_from_vmdb():
         fail_condition=None
     )
     subnet.validate_stats(subnet_stats)
+
+
+def test_network_router_details_stats(appliance, provider, with_nuage_sandbox):
+    """
+    This test creates Nuage enterprise and its entities, including network router.
+    Then it validates inventory of created network router.
+    Steps:
+        * Deploy some entities outside of MIQ (directly in the provider)
+        * Create dictionary with stats that will be validated
+        * Find created network router in database and validate its stats
+    """
+    sandbox = with_nuage_sandbox
+    router_stats = {
+        'name_value': sandbox['domain'].name,
+        'type_value': 'ManageIQ/Providers/Nuage/Network Manager/Network Router',
+        'network_manager_value': provider.name,
+        'cloud_tenant_value': sandbox['enterprise'].name,
+        'cloud_subnets_num': len(sandbox['domain'].subnets.get()),
+        'floating_ips_num': len(sandbox['domain'].floating_ips),
+        'security_groups_num': len(sandbox['domain'].policy_groups)
+    }
+    provider.refresh_provider_relationships()
+
+    def router_from_vmdb():
+        logger.info('Looking for network router with ems_ref {} in the VMDB...'.format(
+            sandbox['domain'].id)
+        )
+        return appliance.collections.nuage_network_routers.find_by_ems_ref(sandbox['domain'].id)
+
+    router, _ = wait_for(
+        router_from_vmdb,
+        num_sec=60,
+        delay=5,
+        fail_condition=None
+    )
```

Comment:
I see we're now copy-pasting L77-88 from subnet, only this time its for tenant. Think we could introduce auxilary function, say

```python
def object_in_vmdb_with_timeout(table, ems_ref):
   ...
```

So we could reuse it for every entity then?

```python
object_in_vmdb_with_timeout('cloud_tenants', 123)
```